control-plane/pkg/reconciler/channel/controller.go

```diff
@@ -110,6 +113,21 @@ func NewController(ctx context.Context, watcher configmap.Watcher, configs *conf
 		impl.GlobalResync(channelInformer.Informer())
 	}
 
+	rotateCACerts := func(obj interface{}) {
+		newCerts, err := reconciler.getCaCerts()
+		if err != nil && (features.IsPermissiveTransportEncryption() || features.IsStrictTransportEncryption()) {
+			// We only need to warn here as the broker won't reconcile properly without the proper certs because the prober won't succeed
+			logger.Warn("Failed to get new CA certs while rotating CA certs when at least one address uses TLS", zap.Error(err))
+		}
+		reconciler.NewProber.RotateRootCaCerts(&newCerts)
+		globalResync(obj)
+	}
+
+	secretinformer.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.FilterWithName(kafkaChannelTLSSecretName),
+		Handler:    controller.HandleAll(rotateCACerts),
+	})
+
```

Comment:
```suggestion

```
We don't need two event handlers with the same FilterFunc, there should be another EventHandler being registered a bit farther down in the file where you can change the Handler from using `globalResync` to using `rotateCACerts`